File: backend/internal/domain/mahasiswa.go
package domain

import "time"

type Mahasiswa struct {
	ID          uint64    `db:"id"             json:"id"`
	UserID      *uint64   `db:"user_id"        json:"user_id,omitempty"`
	NIM         string    `db:"nim"            json:"nim"`
	Nama        string    `db:"nama"           json:"nama"`
	Email       string    `db:"email"          json:"email"`
	Jurusan     string    `db:"jurusan"        json:"jurusan"`
	Semester    int       `db:"semester"       json:"semester"`
	DosenPAID   *uint64   `db:"dosen_pa_id"    json:"dosen_pa_id,omitempty"`
	DosenPANama string    `db:"dosen_pa_nama"  json:"dosen_pa_nama,omitempty"`
	CreatedAt   time.Time `db:"created_at"     json:"created_at"`
	UpdatedAt   time.Time `db:"updated_at"     json:"updated_at"`
}

File: backend-refactor-python/app/schemas/mahasiswa.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class MahasiswaResponse(BaseModel):
    id: int
    user_id: Optional[int]
    nim: str
    nama: str
    email: str
    jurusan: str
    semester: int
    dosen_pa_id: Optional[int]
    dosen_pa_nama: Optional[str]
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class CreateMahasiswaRequest(BaseModel):
    user_id: Optional[int] = None
    nim: str
    nama: str
    email: str
    jurusan: str
    semester: int = 1
    dosen_pa_id: Optional[int] = None


class UpdateMahasiswaRequest(BaseModel):
    nim: Optional[str] = None
    nama: Optional[str] = None
    email: Optional[str] = None
    jurusan: Optional[str] = None
    semester: Optional[int] = None
    dosen_pa_id: Optional[int] = None
